espnet/scheduler/scaler.py

```diff
@@ -0,0 +1,170 @@
+"""Scalers."""
+
+import argparse
+
+from espnet.utils.dynamic_import import dynamic_import
+from espnet.utils.fill_missing_args import fill_missing_args
+
+
+class _PrefixParser:
+    def __init__(self, parser, prefix):
+        self.parser = parser
+        self.prefix = prefix
+
+    def add_argument(self, name, **kwargs):
+        assert name.startswith("--")
+        self.parser.add_argument(self.prefix + name[2:], **kwargs)
+
+
+class ScalerInterface:
+    """Scaler interface."""
+
+    alias = ""
+
+    def __init__(self, key: str, args: argparse.Namespace):
+        """Initialize class."""
+        self.key = key
+        prefix = key + "_" + self.alias + "_"
+        for k, v in vars(args).items():
+            if k.startswith(prefix):
+                setattr(self, k[len(prefix):], v)
+
+    def get_arg(self, name):
+        """Get argument without prefix."""
+        return getattr(self.args, f"{self.key}_{self.alias}_{name}")
+
+    @classmethod
+    def add_arguments(cls, key: str, parser: argparse.ArgumentParser):
+        """Add arguments for CLI."""
+        group = parser.add_argument_group(f"{cls.alias} scaler")
+        cls._add_arguments(_PrefixParser(parser=group, prefix=f"--{key}-{cls.alias}-"))
```

Comment:
Could you tell me why you nest function?

Reply:
Now I got it.
But I'm not sure it is really needed.

Reply:
This is related to your previous question. I just want to support multiple schedules for multiple hyperparameters (as SpecAugument paper did). Therefore, I need to rename args like `--warmup` for `lr` and `weight_decay` to `--lr-warmup` and `--weight_decay-warmup`